src/Columns/ColumnReplicated.cpp

```diff
@@ -0,0 +1,678 @@
+#include <Columns/ColumnReplicated.h>
+#include <Columns/ColumnCompressed.h>
+#include <Columns/ColumnConst.h>
+#include <Common/WeakHash.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int SIZES_OF_COLUMNS_DOESNT_MATCH;
+    extern const int LOGICAL_ERROR;
+}
+
+ColumnReplicated::ColumnReplicated(MutableColumnPtr && nested_column_)
+    : nested_column(std::move(nested_column_))
+{
+    indexes.insertIndexesRange(0, nested_column->size());
```

Comment:
Can we deprecate this ctor? If ColumnReplicated can replace any column, then we can always use the initial one without indexes.